### PR TITLE
Upgrade to pop v6

### PR DESCRIPTION
**Pull request by ahobson**
## Summary

The two major API changes are to the migrator interface and how errors
are returned.

This affects
 * cmd/milmove/migrate.go
 * pkg/handlers/errors.go
 * pkg/migrate/Builder_test.go

I can't seem to find any detailed release notes for v6 other than https://github.com/gobuffalo/pop/releases/tag/v6.0.0

## Setup to Run Your Code

```sh
make server_test
```

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.

### CODE REVIEW

**Review by reggieriser — COMMENTED** [review]
This LGTM -- thanks for handling the upgrade and the fixes for the API changes! 🥇 

A few small things:

- I was wondering if we could go ahead and upgrade to the latest `6.0.2` version while we're at it.
- In `.golangci.yml`, there are some `gomodguard` entries that talk about `pop/v5` that should probably be updated.
- In `testdata/src/github.com/gobuffalo/pop/v6/main.go`, I saw the path changed from `v5` to `v6`, but should we change the package there from `v5` to `v6` as well?  I'm not familiar with this part of the code, so I'm not sure if that package name actually matters (even if it doesn't, though, we should probably sync it or use a non-versioned package name).

**Comment by ahobson** [comment]
> This LGTM -- thanks for handling the upgrade and the fixes for the API changes! 🥇
> 
> A few small things:
> 
> * I was wondering if we could go ahead and upgrade to the latest `6.0.2` version while we're at it.
> * In `.golangci.yml`, there are some `gomodguard` entries that talk about `pop/v5` that should probably be updated.
> * In `testdata/src/github.com/gobuffalo/pop/v6/main.go`, I saw the path changed from `v5` to `v6`, but should we change the package there from `v5` to `v6` as well?  I'm not familiar with this part of the code, so I'm not sure if that package name actually matters (even if it doesn't, though, we should probably sync it or use a non-versioned package name).



> This LGTM -- thanks for handling the upgrade and the fixes for the API changes! 🥇
> 
> A few small things:
> 
> * I was wondering if we could go ahead and upgrade to the latest `6.0.2` version while we're at it.

Yeah! I forgot to `go get` the latest, but I've done so now

> * In `.golangci.yml`, there are some `gomodguard` entries that talk about `pop/v5` that should probably be updated.

Done!

> * In `testdata/src/github.com/gobuffalo/pop/v6/main.go`, I saw the path changed from `v5` to `v6`, but should we change the package there from `v5` to `v6` as well?  I'm not familiar with this part of the code, so I'm not sure if that package name actually matters (even if it doesn't, though, we should probably sync it or use a non-versioned package name).

Thank you! Fixed

File: go.mod [hunk 1]
```diff
@@ -84,6 +83,7 @@ require (
 )
 
 require (
+	github.com/gobuffalo/pop/v6 v6.0.0
```
[the remark this answers is not in the document]
Comment (ahobson) [reply go.mod]: I'm honestly not sure. After reading https://go.dev/ref/mod#graph-pruning I'm still not sure. I did `go get -u github.com/gobuffalo/pop/v6@latest`

**Review by reggieriser — APPROVED** [review]
Thanks for making the updates!  🚢

**Comment by ahobson** [comment]
I deployed this to experimental and AFAICT, all seems well

**Pull request merged**